CONTRIBUTING.md

```diff
@@ -67,6 +67,7 @@ and pre-commit to run some validation checks:
 
 ```shell
 pipx install pre-commit  # or `pip install pre-commit` if you have a virtualenv
+pre-commit install
```

Comment:
I believe the reason why it is not mentioned is because using pre-commit depends on personal preference. I prefer *not* to use pre-commit because I want to be able to commit intermediate states, even if they violate some pre-commit checks. I also find it important that commits are instant, I don't want to wait for slow checks because it interrupts my development workflow. 

However, I find it useful to point out the possibility of using and installing pre-commit in the contributing.md as you did. It may be worth to add a comment making it clear that this step is optional and only necessary if you want to run the checks on every commit.

Reply:
Thanks for the comment. Adding a comment sounds good.

Reply:
Maybe, suggest using `pre-push` instead of `pre-commit`. Like this, commit are instant, only when you want to push will it run pre-commit and avoid running CI for invalid code.

Reply:
You kinda do need `pipx install pre-commit`, e.g. for `pre-commit run --all-files`, so i'd split this blog into two sections: One for `pipx install pre-commit` and one for the optional `pre-commit install`